README.md

```diff
@@ -1,46 +1,41 @@
-## gem5 in docker
-### What does it do?
-The main branch sets up the ROCm environment up to HIP using .deb files available on repo.radeon.com/rocm/archive.
-
-The machine learning branch sets up the ROCm environment up to HCC. The user then needs to build the remaining packages to allow machine learning programs to run. 
-
-### Build
-Clone the repo using `git clone --recursive https://github.com/KyleRoarty/gem5_docker.git`\
-Apply the patches in the `patch` directory to the submodule with the same name\
-Build the docker image by running `docker build --build-arg rocm_ver=<version> -t gem5 .` creating a docker image "gem5".\
-This downloads the archive version specified in the build command (or 1.6.0 by default) from repo.radeon.com/rocm/archive, and installs:\
-**Master**
-* hsakmt-roct-dev
-* hsa-ext-rocr-dev
-* hsa-rocr-dev
-* rocm-utils
-* hcc
-* hip_base
-* hip_hcc
-* hip_samples
-
-**Machine Learning**
-* hsakmt-roct-dev
-* hsa-ext-rocr-dev
-* hsa-rocr-dev
-* rocm-utils
-* hcc
-
-These are installed to /opt/rocm, and environment variables used in ROCm (ROCM_PATH, HCC_HOME, HSA_PATH, HIP_PLATFORM) are set.
-
-Start a container:\
-`docker run --name <container_name> -it -d -v/gem5/parent/dir:/sim/ gem5`\
-Mapping the parent directory of gem5 to /sim/ in the docker container
-
-**Machine Learning**\
-Build the remaining dependencies manually. The commands to do so are found in cmd/deps.txt
-
-
-Build gem5 in the docker container:\
-`docker exec -w/sim/gem5 <container_name> scons -jN /sim/gem5/build/GCN3_X86/gem5.opt`
-
-To run something in gem5:\
-`docker exec <container_name> /sim/gem5/build/GCN3_X86/gem5.opt /sim/gem5/configs/example/apu_se.py -n2 -c<command> --options="<command options>"`  
-
-### Something isn't working?
-* A program may be missing a dependency: `docker exec <container_name> apt-get install -y <package>`
+## gem5 in docker for machine learning
+### What is it?
+This builds a docker image that contains gem5, ROCm, and MIOpen (using HIP). It's intended for running machine learning programs in the APU model in the GCN3 branch of gem5.
+
+The container uses ROCm 1.6.2; This can be overridden when building the docker image.
+
+### How to build/run?
+[The dockerhub repo](https://cloud.docker.com/repository/registry-1.docker.io/kroarty/gem5) Will contain an image for this branch soon. Currently it only has the non-ML-enabled image.
+
+From source:
+```
+docker build [--build-arg rocm_ver=<version>] -t <im_name> .
+docker run [--name <container_name>] -it [-d] <im_name> [<command>]
+```
+If '-d' is specified, the container will run in the background \
+If '\<command\>' isn't specified, the container will run a bash shell
+
+### Building gem5 in the container
+
+To keep filesize down, gem5 is not pre-built. So on the initial run of a container, gem5 needs to be built.
```

Comment:
Practically speaking, what is the implication of this?  Is it possible to build once on CHTC with this?  Or would Condor need to simulate the building every time?

Reply:
As far as filesize, it's a difference between downloading a 2GB file and an 8GB file. I don't know how docker hub handles changes and if the file has to be completely re-downloaded every change.

It is possible to download the image, build gem5, and then commit that as a new docker image. 

The [CHTC documentation](http://chtc.cs.wisc.edu/docker-jobs.shtml#submit) says it pulls from docker hub every time, but I don't know if docker images are cached at all. As far as I can tell, it would have to rebuild gem5 every time.

Reply:
Got it.  We'll have to think if there is a more general solution.